csrc/scheduler/matmul.cpp

```diff
@@ -646,16 +713,44 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
     }
   }
 
-  // [Mo, No, Ko, Mi, Ni, Ki]
-  // Propagate tiling globally
-  scheduler_utils::transformPropagateToAllFrom(cc, -1);
+  if (params.split_k_factor > 1) {
+    // [Mo, No, Ko, Mi, Ni, Ki]
+    constexpr int Ko_axis = 2;
+    cc->split(Ko_axis, params.split_k_factor, false);
+    // Propagate tiling to all except the output tensor, which will be scheduled
+    // differently
+    const std::vector<TensorView*>& selected_tvs =
+        ir_utils::allTvsExcept(fusion, {c});
+    scheduler_utils::transformPropagateToSelectedFrom(
+        cc, {selected_tvs.begin(), selected_tvs.end()});
+  } else {
+    // [Mo, No, Ko, Mi, Ni, Ki]
+    // Propagate tiling globally
+    scheduler_utils::transformPropagateToAllFrom(cc, -1);
+  }
 
   // Schedule warp tile
   mma_utils::scheduleWarpTileWithReduction(cc, gemm_tile);
 
-  // Propagate warp tile to main loop and epilog/output tvs
-  scheduler_utils::BoundedDirectionalTransformPropagator::bothWays(
-      cc, -1, {acw_smem, bcw_smem}, {c});
+  // mma_result -> tmp_gmem[g] -> tmp_gmem_reload -> cc -> c
+  TensorView* mma_result = cc; // mma results
+  TensorView* tmp_gmem = nullptr; // output
+  TensorView* tmp_gmem_reload = nullptr;
+  if (params.split_k_factor > 1) {
+    // mma_result -> tmp_gmem[g] -> tmp_gmem_reload -> cc -> c[output]
+    tmp_gmem = cc->rFactor({3, 4, -1});
+    mma_result = tmp_gmem->cacheBefore();
+    tmp_gmem->setMemoryType(MemoryType::Global);
+    tmp_gmem_reload = tmp_gmem->cacheAfter();
+    mma_builder.accumulatorTv(mma_result);
```

Comment:
Can you move the `mma_builder.accumulatorTv(cc);` before down to the `else` clause? This way it is easier to understand.